### PR TITLE
Feature/link takepicture words

### PR DESCRIPTION
**Pull request by rombirli**

### CODE REVIEW

File: app/src/main/java/com/github/sdpsharelook/language/Language.kt [hunk 1]
```diff
@@ -2,11 +2,12 @@ package com.github.sdpsharelook.language
...
 import android.content.Context
 import com.github.sdpsharelook.R
+import java.io.Serializable
 import java.util.*
 
 data class Language(
-    val tag: String
-) {
+    val tag: String,
+) : Serializable {
```
Comment (AdrianUlrich) [inline app/src/main/java/com/github/sdpsharelook/language/Language.kt:10]: Use kotlin [serialization](https://kotlinlang.org/docs/serialization.html)

**Review by AdrianUlrich — APPROVED** [review]
Very good job deleting the dialogs and dealing with permissions
If you have time use the correct form of serialization
Also it would be good if @albastardoto and @edoulaX could review the Word-SectionWord business
Also I don't know why but the button translate to section is not working anymore here, so maybe fix it unless that was intended, which is fine

File: app/src/main/java/com/github/sdpsharelook/translate/TranslateFragment.kt [hunk 2]
```diff
@@ -108,7 +108,7 @@ open class TranslateFragmentLift : Fragment() {
         button: Button,
         translatorLanguages: Set<Language>,
         buttonSource: Button,
-        buttonTarget: Button
+        buttonTarget: Button,
```
Comment (Sonusiety) [inline app/src/main/java/com/github/sdpsharelook/translate/TranslateFragment.kt:111]: Should remove unnecessary comma

File: app/src/main/java/com/github/sdpsharelook/speechRecognition/SpeechRecognizer.kt [hunk 3]
```diff
@@ -1,81 +1,55 @@
...
-                activity,
-                android.Manifest.permission.RECORD_AUDIO
-            ) != PackageManager.PERMISSION_GRANTED
-        ) {
-            requestPermissionLauncher.launch(
-                android.Manifest.permission.RECORD_AUDIO
-            )
-        }
-    }
+    private val speechRecognizer = GoogleSpeechRecognizer.createSpeechRecognizer(context)
```
Comment (Sonusiety) [inline app/src/main/java/com/github/sdpsharelook/speechRecognition/SpeechRecognizer.kt:52]: Good job extracting permissions from the camera class.

**Review by Sonusiety — APPROVED** [review]

**Pull request merged**